clang/lib/CodeGen/CGExprComplex.cpp

```diff
@@ -761,6 +765,25 @@ ComplexPairTy ComplexExprEmitter::EmitBinMul(const BinOpInfo &Op) {
 
     CodeGenFunction::CGFPOptionsRAII FPOptsRAII(CGF, Op.FPFeatures);
     if (Op.LHS.second && Op.RHS.second) {
+      if (Op.FPFeatures.getComplexRange() == LangOptions::CX_Limited ||
+          (Op.FPFeatures.getComplexRange() == LangOptions::CX_None &&
+           CGF.getLangOpts().getComplexRange() == LangOptions::CX_Limited) ||
+          (Op.FPFeatures.getComplexRange() == LangOptions::CX_None &&
+              CGF.getLangOpts().getComplexRange() == LangOptions::CX_Fortran) ||
+          CGF.getLangOpts().FastMath) {
+        // (a+ib)*(c+id) = (ac-bd)+i(bc+ad)
```

Comment:
I'm not in agreement with this conditional:

* I believe pragma should generally override command-line options
* fast-math should probably handled in the driver by setting default complex range pragma settings.